packages/dynamics-lib/src/entities/__tests__/pocl-validaton-error.entity.spec.js

```diff
@@ -52,6 +53,41 @@ describe('pocl staging exception entity', () => {
     it('has the expected data', () => {
       expect(exception).toMatchSnapshot()
     })
+
+    it('has the expected fields', () => {
+      const expectedFields = {
+        firstName: 'Daniel',
+        lastName: 'Ricciardo',
+        organisation: 'Fishy Endeavours',
+        premises: '14 Howecroft Court',
+        street: 'Eastmead Lane',
+        locality: 'Stoke Bishop',
+        town: 'Bristol',
+        postcode: 'BS9 1HJ',
+        country: 'GB',
+        birthDate: '1989-07-01',
+        email: 'daniel-ricc@example.couk',
+        mobilePhone: '07722 123456',
+        preferredMethodOfConfirmation: expect.objectContaining({ id: 910400000, label: 'Email', description: 'Email' }),
+        preferredMethodOfNewsletter: expect.objectContaining({ id: 910400003, label: 'Prefer not to be contacted', description: 'Prefer not to be contacted' }),
+        preferredMethodOfReminder: expect.objectContaining({ id: 910400002, label: 'Text', description: 'Text' }),
+        concessions: '[{"type":"Blue Badge","referenceNumber":123456789}]',
+        startDate: '2021-06-15',
+        serialNumber: '14345-48457J',
+        permitId: 'test-permit-id',
+        transactionDate: '2020-01-01T14:00:00Z',
+        amount: 30,
+        paymentSource: 'Post Office Sales',
+        channelId: '948594',
+        methodOfPayment: expect.objectContaining({ id: 910400001, label: 'Cash', description: 'Cash' }),
+        status: expect.objectContaining({ id: 910400000, label: 'Needs Review', description: 'Needs Review' }),
+        dataSource: expect.objectContaining({ id: 910400000, label: 'Post Office Sales', description: 'Post Office Sales' }),
+        stateCode: 1,
+        errorMessage: '\"permissions[0].licensee.email\" must be a valid email'
+      }
+
+      expect(exception).toMatchObject(expect.objectContaining({ etag: 'W/"56351087"', ...expectedFields }))
```

Comment:
I'm a bit confused how the snapshot doesn't cover this...?

Reply:
It does, but SonarCloud doesn't seem to count the snapshot as part of its coverage, so it keeps flagging this file as not having enough test coverage without this added test